src/librustc/middle/def.rs

```diff
@@ -29,8 +29,9 @@ pub enum Def {
     DefAssociatedConst(DefId),
     DefLocal(DefId, // def id of variable
              ast::NodeId), // node id of variable
-    DefVariant(DefId /* enum */, DefId /* variant */, bool /* is_structure */),
-    DefTy(DefId, bool /* is_enum */),
+    DefVariant(DefId /* enum */, DefId /* variant */),
```

Comment:
Since you’re refactoring here, might it also be a good time to move from tuple-variants to struct-variants, so comments like these aren’t necessary?